### PR TITLE
Upgrade v0.9 instructions

### PR DESCRIPTION
**Pull request by tpltnt**
Hi there,

I documented the upgrading process (to v0.9).

Cheers,
tpltnt

### CODE REVIEW

**Comment by freesig** [comment]
Hey thanks for the PR.
Just a few things.
1. With the dependencies on linux you only want the ones specific to your actual card. ie don't install radeon if you have nvidia etc. Also mesa is really only for Intel. 
Also there is the issue of wether or not to use the official Nvidia drivers for ubuntu or the PPA ones. The official ones are much more recent and stable but they are a major PITA to install. What are your thoughts on the @mitchmindtree ? I'd love to support as many OS's automatic vulkan installation as possible but it's kinda hard.
2. I think we want to encourage people to get new versions of nannou through crates.io not by cloning the github. Most people will be using nannou as a dependacy and all they need to do to update is change `nannou = "0.8"` to `nannou = "0.9"`. This also allows us to push out minor releases without a major update.
Basically a nannou user shouldn't need to use `git`, they should be able to do everything through `cargo`.
We could have a separate section for nannou developers with instructions on how to contribute etc.

**Comment by tpltnt** [comment]
I have split the instructions.

File: src/getting_started/upgrading.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+You can upgrade to a new version of nannou by editing your `Cargo.toml` file to use the new
+crate. For v0.9 use the line `nannou = "0.9"`. Run
+```bash
+cargo update
+```
+inside the nannou directory to upgrade all dependencies.
+
+Building Nannou examples might still fail. This is most likely due to new language features.
+Nannou 0.9 for example requires rustc 1.35.0.
+You can upgrade you (local) rust toolchain by executing
```
Comment (mitchmindtree) [inline src/getting_started/upgrading.md]: I think this should be "You can upgrade **your**" (not you)

File: src/getting_started/platform-specific_setup.md [hunk 2]
```diff
@@ -54,6 +54,67 @@ Depending on what OS you are running, you might require an extra step or two.
     [others](https://en.wikipedia.org/wiki/LibreSSL#Adoption) if manually
     installed).
 
+  - **vulkan**
+
```
Comment (mitchmindtree) [inline src/getting_started/platform-specific_setup.md:58]: Perhaps it's worth adding the following text here before guiding users to the commands below:

> Installing Vulkan support on Linux is generally quite easy using your distro's package manager. That said, there may be different driver options to consider depending on your graphics card and tolerance for proprietary software. The following are rough guidelines on how to get going quickly, however if you are at all concerned with finding the approach that suits you best we recommend searching for vulkan driver installation for your graphics card on your distro.

cc @freesig would this address your concerns re the possibility for different drivers? I honestly don't think we'll ever be able to feasibly automate Linux installation entirely due to the number of everchanging distros, but I also think Linux users are generally used to adding a couple of commands like this anyways. Maybe we can revisit the possibility of automating it on at least some distros if we have a go at a graphical installer in the future?

**Review by mitchmindtree — CHANGES_REQUESTED** [review]
Great work @tpltnt, thanks a lot for doing this!

For the most part it looks good to me, I left a couple of comments on the code for things that could be addressed.

File: src/getting_started/platform-specific_setup.md (hunk 2 above)
Comment (tpltnt) [reply src/getting_started/platform-specific_setup.md:58]: added the text

File: src/getting_started/upgrading.md (hunk 1 above)
Comment (tpltnt) [reply src/getting_started/upgrading.md]: typo fixed

**Comment by tpltnt** [comment]
all requested changes are done

**Review by mitchmindtree — APPROVED** [review]
